delta/sourcing/src/main/scala/ch/epfl/bluebrain/nexus/sourcing/projections/ProjectionProgress.scala

```diff
@@ -25,9 +26,14 @@ final case class ProjectionProgress(
     */
   def +(message: Message[_]): ProjectionProgress =
     message match {
-      case _: DiscardedMessage => copy(offset = message.offset, processed = processed + 1, discarded = discarded + 1)
-      case _: ErrorMessage     => copy(offset = message.offset, processed = processed + 1, failed = failed + 1)
-      case _                   => copy(offset = message.offset, processed = processed + 1)
+      case _: DiscardedMessage  => copy(offset = message.offset, processed = processed + 1, discarded = discarded + 1)
+      case _: ErrorMessage      => copy(offset = message.offset, processed = processed + 1, failed = failed + 1)
+      case s: SuccessMessage[_] =>
+        copy(
+          offset = message.offset,
+          warnings = warnings + s.warnings.headOption.fold(0)(_ => 1),
```

Comment:
I'm a bit confused. If the passed `SuccessMessage` has several warning messages we still add just one?

Reply:
Yep, you are right, I kept in mind to increment like the others but it makes less sense